src/PPN-microbench/cpu_frequency.cpp

```diff
@@ -1,50 +1,77 @@
 #include <PPN-microbench/cpu_frequency.hpp>
 
-CPUFrequency::CPUFrequency(std::string name, int nbIterations) : Microbench(name, nbIterations)  {
-    nbThreads = std::thread::hardware_concurrency() - 1;
-    measures.reserve(nbThreads * nbIterations);
-    benchTimes.reserve(nbThreads * nbIterations);
+CPUFrequency::CPUFrequency(int nbMeasures) : Microbench("CPU Frequency", 9999999){
+    this->nbMeasures = nbMeasures;
+    Context context = Context::getInstance();
+    nbCores = context.getCpus();
+    measures = std::make_unique<double[]>(nbMeasures * ((nbCores * (nbCores + 1)) / 2));
 }
 
 CPUFrequency::~CPUFrequency() {}
 
-void CPUFrequency::executeBench(int id) {
-    u64 startMeasure = rdtsc();
-    std::chrono::steady_clock::time_point startTime = std::chrono::steady_clock::now();
-    int cpt;
-    for (int i = 0; i < INT32_MAX; i++) {
-        cpt++;
+void CPUFrequency::executeAdds() {
+    int cpt = 0;
+    for (int i = 0; i < getNbIterations(); i++) {
+        // 16 adds
+        cpt++; cpt++; cpt++; cpt++; cpt++; cpt++; cpt++; cpt++;
```

Comment:
Why remove the loop?
```suggestion
        for (size_t _ = 0; _ < 16; ++_) { cpt++; }
```

Reply:
Won't this cause problems when compiling in O3 and just remove the line?